pkg/dynamicinstrumentation/eventparser/event_parser.go

```diff
@@ -21,122 +22,109 @@ import (
 	"github.com/DataDog/datadog-agent/pkg/util/log"
 )
 
-// MaxBufferSize is the maximum size of the output buffer from bpf which is read by this package
-const MaxBufferSize = 10000
-
 var (
 	byteOrder = binary.LittleEndian
 )
 
 // ParseEvent takes the raw buffer from bpf and parses it into an event. It also potentially
 // applies a rate limit
-func ParseEvent(record []byte, ratelimiters *ratelimiter.MultiProbeRateLimiter) *ditypes.DIEvent {
+func ParseEvent(procs ditypes.DIProcs, record []byte, ratelimiters *ratelimiter.MultiProbeRateLimiter) (*ditypes.DIEvent, error) {
 	event := ditypes.DIEvent{}
 
 	if len(record) < ditypes.SizeofBaseEvent {
-		log.Tracef("malformed event record (length %d)", len(record))
-		return nil
+		return nil, fmt.Errorf("malformed event record (length %d)", len(record))
 	}
 	baseEvent := *(*ditypes.BaseEvent)(unsafe.Pointer(&record[0]))
 	event.ProbeID = unix.ByteSliceToString(baseEvent.Probe_id[:])
 
 	allowed, droppedEvents, successfulEvents := ratelimiters.AllowOneEvent(event.ProbeID)
 	if !allowed {
-		log.Tracef("event dropped by rate limit. Probe %s\t(%d dropped events out of %d)\n",
+		return nil, log.Errorf("event dropped by rate limit. Probe %s\t(%d dropped events out of %d)\n",
```

Comment:
The error is logged at the trace level (for normal events, error level for config), do you think that's too noisy still?

Reply:
> The error is logged at the trace level (for normal events, error level for config), do you think that's too noisy still?

I'm confused, maybe I don't understand what you mean 🙂 I see that:

* we are replacing Tracef with Errorf here
* this is the generic event parser, not for config probe events
* log.Errorf will log an error

Am I missing something?

We should not be rate limiting the config probe at all.

Reply:
Ah ok I'm confused because I accidentally made this returning `log.Errorf`, I meant to make it `fmt.Errorf`. Thanks for catching that. Now that i've changed it my above comment is accurate. We do not rate limit in config probe, the error is still checked though.